app/services/state.py

```diff
@@ -23,8 +23,9 @@ async def get_state(self, integration_id: str, action_id: str, source_id: str =
     async def set_state(self, integration_id: str, action_id: str, state: dict, source_id: str = "no-source"):
         for attempt in stamina.retry_context(on=redis.RedisError, attempts=5, wait_initial=1.0, wait_max=30, wait_jitter=3.0):
             with attempt:
-                await self.db_client.set(
+                await self.db_client.setex(
                     f"integration_state.{integration_id}.{action_id}.{source_id}",
+                    7*86400,
```

Comment:
Usually, the state is persistent but I see we want it to expire after some days here. I'd like to know more about this use case so we can add proper support for it in the template.